src/utils/access_token.ts

```diff
@@ -0,0 +1,119 @@
+import { interruptSpinner, wait } from "./spinner.ts";
+import { error } from "../error.ts";
+import { endpoint } from "./api.ts";
+import tokenStorage from "./token_storage.ts";
+
+export default {
+  get: tokenStorage.get,
+
+  async provision() {
+    // Synchronize provision routine
+    // to prevent multiple authorization flows from triggering concurrently
+    this.provisionPromise ??= provision();
+    const token = await this.provisionPromise;
+    this.provisionPromise = null;
+    return token;
+  },
+  provisionPromise: null as Promise<string> | null,
+
+  revoke: tokenStorage.remove,
+};
+
+async function provision(): Promise<string> {
+  const spinnerInterrupted = interruptSpinner();
+  wait("").start().info("Provisioning a new access token...");
```

Comment:
Not really sure how spinner exactly works, but what `wait("").start().info("some message")` does is basically:

1. starting the spinner with no message along the spinner
2. then stopping (and clearing) the spinner with an info message printed

Is this correct? If correct, then starting the spinner will have essentially no effect because it's immediately stopped and cleared - I wonder if we could just use `console.info`

Reply:
The beauty of it is that the output is consistent with the rest of spinner result messages (the info/warn icon, the identation, etc).

Reply:
(this pattern is used everywhere in deployctl)